python/cudf/cudf/core/buffer/weakrefable_buffer.py

```diff
@@ -0,0 +1,324 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+from __future__ import annotations
+
+import copy
+import weakref
+from typing import Any, Dict, Tuple, Type, TypeVar
+
+import rmm
+
+import cudf
+from cudf.core.buffer.buffer import Buffer, cuda_array_interface_wrapper
+
+T = TypeVar("T", bound="RefCountableBuffer")
+
+
+class CachedInstanceMeta(type):
```

Comment:
Given that this is only ever used in one place, I think it's fine for the name to be awkwardly long:

```suggestion
class _CachedWeakrefableBufferInstanceMeta(type):
```